src/Core/src/Hosting/MauiAppBuilder.cs

```diff
@@ -148,6 +149,7 @@ public MauiApp Build()
 			_host.RunDeferredCallbacks(_hostBuilder);
 
 			_builtApplication = new MauiApp(_hostBuilder.Build());
+			_builtApplication.Services.GetRequiredService<HostedServiceExecutor>().StartAsync(default);
```

Comment:
I'm not sure why this is necessary. Why isn't the host created by `_hostBuilder.Build()` running any registered `IHostedService`s?

This is too early to run hosted services anyway. There's no way to pass in a `CancellationToken` to `Build()` and any exceptions thrown from `IHostedService.StartAsync()` will not be observed since the Task is being thrown away. Hosted services should be started by the call to `_host.StartAsync()`  inside `MauiApp.StartAsync()`.

With `WebApplication` which is similar to `MauiApp` we had an issue where custom hosted services were running after the [GenericWebHostedService](https://github.com/dotnet/aspnetcore/blob/7672ca91ab13b0a9f0f94388ddda79e8642831d4/src/Hosting/Hosting/src/GenericHost/GenericWebHostedService.cs) instead of before. This meant you could start receiving requests before all the custom hosted services ran, but that was fixed by https://github.com/dotnet/aspnetcore/pull/36122.

Reply:
@halter73 Ok agree, will commit new changes soon.